### PR TITLE
TT-1473 Refactor integration-tests.yml workflow to use run-e2e-tests-reusable-workflow.yml

### PR DESCRIPTION
**Pull request by lukaszcl**
This PR refactors .github/workflows/integration-tests.yml to use .github/workflows/run-e2e-tests-reusable-workflow.yml for running tests from the core/integration-tests repository. Additionally, it removes citool from this repository following its migration to the CTF repository.

To do: 
- [x] Resolve conflicts and migrate CCIP tests
- [x] Merge and use https://github.com/smartcontractkit/chainlink-testing-framework/pull/1077 
- [x] Make sure tests in remote runner still work

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -4,8 +4,9 @@ on:
   merge_group:
   pull_request:
   push:
-    tags:
-      - "*"
+    # TODO: uncomment when ready
```
Comment (kalverra) [inline .github/workflows/integration-tests.yml]: This just for PR draft?

File: .github/workflows/run-e2e-tests-reusable-workflow.yml [hunk 2]
```diff
@@ -193,22 +225,26 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
       - name: Setup Go
-        uses: ./.github/actions/setup-go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version: '1.22.6'
```
Comment (kalverra) [inline .github/workflows/run-e2e-tests-reusable-workflow.yml:236]: We've had issues before setting a Go version like this (I forget which exactly, but I know our custom `setup-go` was designed to mitigate them). Do we no longer anticipate them?

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (lukaszcl) [reply .github/workflows/integration-tests.yml]: yeah, let me uncomment it

File: .github/workflows/run-e2e-tests-reusable-workflow.yml (hunk 2 above)
Comment (lukaszcl) [reply .github/workflows/run-e2e-tests-reusable-workflow.yml:236]: @kalverra I think our setup-go action works exactly the same in this case, see `https://github.com/smartcontractkit/chainlink-github-actions/blob/main/chainlink-testing-framework/setup-go/action.yml#L37-L43` and https://github.com/actions/setup-go?tab=readme-ov-file#check-latest-version and I don't need our custom caching logic here

**Review by gheorghestrimtu — DISMISSED** [review]

**Review by skudasov — DISMISSED** [review]

**Review by scheibinger — DISMISSED** [review]

**Review by gheorghestrimtu — DISMISSED** [review]

**Review by skudasov — APPROVED** [review]

**Review by gheorghestrimtu — APPROVED** [review]

**Pull request merged**